### DISCUSSION TITLE
May be an infinite loop occurs during the ossl_rsa_fips186_4_gen_prob_primes()?

### DISCUSSION THREAD

**Discussion by liwei62** [openssl/openssl#24777] [Q&A] [answered]
After the ossl_rsa_fips186_4_gen_prob_primes() generates p and q according to the FIPS standard, it checks whether the |p - q| & |Xp - Xq| is valid. It needs to generate q again if its not satisfied the relationship mentioned above. But we find that the input 'cb' is not used in here to check if we need to keep trying to generate q. Could there be an infinite loop here?

**Comment by t8m** [comment, marked as answer]
I do not think there can ever be an infinite loop unless the rng is stuck. The probability of generating p and q values that would not satisfy the condition after a few loops would be so negligible that it in practice cannot happen. However adding a call to the callback inside the loop would make sense. If you want to add it please submit a pull request with such change.
